limitQueryAmountMiddleware.go

```diff
@@ -0,0 +1,75 @@
+package graphql
+
+import (
+	"context"
+
+	gql "github.com/99designs/gqlgen/graphql"
+	"github.com/vektah/gqlparser/v2/ast"
+)
+
+const (
+	sameOperationsDefaultThreshold = 2
+	allOperationsDefaultThreshold  = 10
+)
+
+func LimitQueryAmountMiddleware(
```

Comment:
Since its both top-level queries and mutations lets go for operations:
```suggestion
func LimitOperationAmountMiddleware(
```